src/main.cpp

```diff
@@ -6794,6 +6793,107 @@ ThresholdState VersionBitsTipState(const Consensus::Params& params, Consensus::D
     return VersionBitsState(chainActive.Tip(), params, pos, versionbitscache);
 }
 
+bool LoadMempool(void)
+{
+    FILE* filestr = fopen((GetDataDir() / "mempool.dat").c_str(), "r");
+    CAutoFile file(filestr, SER_DISK, CLIENT_VERSION);
+    if (file.IsNull()) {
+        return error("Failed to open mempool file from disk");
+    }
+
+    int64_t count = 0;
+
+    try {
+        uint64_t version;
+        file >> version;
+        if (version != 1) {
+            return false;
+        }
+        uint64_t num;
+        file >> num;
+        double prioritydummy = 0;
+        while (num--) {
+            CTransaction tx;
+            int64_t nTime;
+            int64_t nFeeDelta;
+            file >> tx;
+            file >> nTime;
+            file >> nFeeDelta;
+
+            CAmount amountdelta = nFeeDelta;
+            if (amountdelta) {
+                mempool.PrioritiseTransaction(tx.GetHash(), tx.GetHash().ToString(), prioritydummy, amountdelta);
+            }
+            CValidationState state;
+            AcceptToMemoryPool(mempool, state, tx, true, NULL, nTime, 0, false);
```

Comment:
Seems to require holding `cs_main`.
`AcceptToMemoryPoolWorker` has a `AssertLockHeld(cs_main);`